server/lib/chat/chat-service.ts

```diff
@@ -115,7 +114,7 @@ export default class ChatService {
       .on('userQuit', userId => this.handleUserQuit(userId))
   }
 
-  private updateUserAfterJoining(
+  private async updateUserAfterJoining(
```

Comment:
If you change this method to async, you need to be changing every single place that calls this. If your intention is that this method could never fail, you'd be better off not making this async but instead wrapping the contents with `Promise.resolve().then(async () => { /* ... */ }).catch(err => { /* do whatever error handling */ })`

In this case though, I think it's a pretty tough argument to make that the error handling should always be generic, it seems like the things calling this should decide how they want to handle failures